kalamari/kalamari.py

```diff
@@ -16,14 +17,30 @@ def __init__(self, json):
                 self.json = Tree.tree_from_dict(loads(json))
 
     def __repr__(self):
-        result = {}
-        for node in self.json[1]:
-            if node.children:
-                result[node.data] = "{...}"
-            if node.container:
-                result[node.data] = "..."
+        try:
+            result = {}
+            for node in self.json[1]:
+                if node.children:
+                    result[node.data] = "{...}"
+                if node.container:
+                    result[node.data] = "..."
+        except KeyError:
+            pass
         return str(result)
 
+    def __iter__(self):
+        try:
+            for i in self.json[1]:
+                yield (i.data, i.get_value())
+        except KeyError:
+            pass
```

Comment:
`self.json[1]` only returns the nodes from the second level (first level is root) of the BFS tree. We would want to iterate through all nodes in all levels, starting from root. I would suggest making this `__iter__` method similar to that of `Tree`'s. The idea is to turn a `smartJSON` object into a `dict`. The underlying elements in the `dict` don't necessarily have to be native Python types like `str` or `list`. So I would just return the original `Node` instance since you can navigate your way through them with methods such as `get_parent()`, `get_children()` etc. We don't want to lose all that tree information